### ISSUE TITLE
With timeout: Uncaught TypeError: request.cancel is not a function

### ISSUE THREAD

**Issue by hedii** [vuejs/vue-resource#310]
When a timeout is set, and when the requested url does not exists or the target server is offline, the request is cancelled with an error:
```js
import Vue from 'vue'
import VueResource from 'vue-resource'

Vue.use(VueResource)

Vue.http.get('http://does-not-exist-or-is-offline.com/something', { timeout: 3000 })
  .then(response => {
    console.log(response)
  }, response => {
    console.log('error')
  })
```

It returns `Uncaught TypeError: request.cancel is not a function` and then `error`

Is it a bug or is there a way to handle this situation?

(UPDATE: i am using ^0.9.1)

**Comment by michaelhue** [comment]
Seems to be related to changes in version 0.9.0.

> Renamed method Request.cancel() to Request.abort()

https://github.com/vuejs/vue-resource/commit/2f951df7261a5bd940438addf83b2a13de06c873#diff-c825ede355666a78dcca43e2d5bf33e5R745

**Comment by hedii** [comment]
@michaelhue yes https://github.com/vuejs/vue-resource/blob/master/src/http/interceptor/timeout.js#L11

maybe it should be `request.abort();`

**Comment by colshacol** [comment]
[This file](https://github.com/vuejs/vue-resource/blob/b9c195daca50188f9338d0900654e75fb6eed483/dist/vue-resource.js) does not reflect the `cancel()` to `abort()` change on the timeout.

**Comment by colshacol** [comment]
I am seeing `request.abort()` defined in `function xhrClient` and `function xdrClient`, but that is it.

@hedii Maybe try going into your local Vue source files and locate this:

```
if (request.timeout) {
    timeout = setTimeout(function () {
        // change this to abort();
        request.cancel();
    }, request.timeout);
}
```

Then report back.

**Comment by hedii** [comment]
closed in #311
